Bottom: 0cf6da0b68b2f5ae61909a53bcfcd9145d6d6789
Top:    53672d19db195ad7b11d4348079b9d517f9483fa
Author: Robert Collins <robertc@robertcollins.net>
Date:   2013-04-02 07:59:41 +1300

DecorateTestCaseResult was missing some used attributes in its exclusion of
forwarded attributes.


---

diff --git a/testtools/testcase.py b/testtools/testcase.py
index 1121ad2..628ddf2 100644
--- a/testtools/testcase.py
+++ b/testtools/testcase.py
@@ -906,8 +906,8 @@ class DecorateTestCaseResult(object):
         delattr(self.decorated, name)
 
     def __setattr__(self, name, value):
-        if name == 'decorated':
-            self.__dict__['decorated'] = value
+        if name in ('decorated', 'callout', 'before_run', 'after_run'):
+            self.__dict__[name] = value
             return
         setattr(self.decorated, name, value)
